Data/Graph/Inductive/Monad.hs

```diff
@@ -19,6 +19,10 @@ module Data.Graph.Inductive.Monad(
 
 
 import Data.Graph.Inductive.Graph
+#if MIN_VERSION_base(4,12,0)
+import Control.Monad.Fail
```

Comment:
recommended way is to depend on `fail` package, then you can provide uniform interface across different GHC versions

- http://hackage.haskell.org/package/fail
- https://ghc.haskell.org/trac/ghc/wiki/Migration/8.0#base-4.9.0.0

---

In general https://ghc.haskell.org/trac/ghc/wiki/Migration/8.6 and other migration guides are good reads.

---

Note that changing context of a class is breaking change, requiring a major bump to `fgl-5.7`